api/v1alpha1/reclaimspacecronjob_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ReclaimSpaceJobTemplateSpec describes the data a Job should have when created from a template
+type ReclaimSpaceJobTemplateSpec struct {
+	// Standard object's metadata of the jobs created from this template.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the job.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +kubebuilder:validation:Required
+	Spec ReclaimSpaceJobSpec `json:"spec,omitempty"`
+}
+
+// ReclaimSpaceCronJobSpec defines the desired state of ReclaimSpaceJob
+type ReclaimSpaceCronJobSpec struct {
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=.+
+	Schedule string `json:"schedule"`
+
```

Comment:
```suggestion
	// Optional deadline in seconds for starting the job if it misses scheduled
	// time for any reason.  Missed jobs executions will be counted as failed ones.
	// +kubebuilder:validation:Optional
	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`

```

if not specified, no deadline will be considered. 

if specified, a job will not be started after (scheduled time + startingdeadlingseconds).
https://pkg.go.dev/k8s.io/api/batch/v1#CronJobSpec

Reply:
ok, I'll add that